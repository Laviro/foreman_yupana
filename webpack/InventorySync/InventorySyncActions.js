import API from 'foremanReact/API';
import { inventorySyncUrl } from './InventorySyncHelpers';
import {
  INVENTORY_SYNC_SUCCESS,
  INVENTORY_SYNC_FAILURE,
} from './InventorySyncConstants';

export const syncInventory = () => async dispatch => {
  try {
    await API.post(inventorySyncUrl('tasks'));
    dispatch({
      type: INVENTORY_SYNC_SUCCESS,
      payload: {},
    });
  } catch (error) {
    dispatch({
      type: INVENTORY_SYNC_FAILURE,
      payload: {
        error: error.message,
      },
    });
  }
};
