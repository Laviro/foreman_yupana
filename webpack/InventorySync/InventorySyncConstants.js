import { translate as __ } from 'foremanReact/common/I18n';

export const INVENTORY_SYNC_SUCCESS = 'INVENTORY_SYNC_SUCCESS';
export const INVENTORY_SYNC_FAILURE = 'INVENTORY_SYNC_FAILURE';

export const INSIGHTS_SYNC_PAGE_TITLE = __('Red Hat Inventory sync');
