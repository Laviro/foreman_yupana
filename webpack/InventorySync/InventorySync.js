import React from 'react';
import { IntlProvider } from 'react-intl';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';
import { Button, Icon } from 'patternfly-react';
import { INSIGHTS_SYNC_PAGE_TITLE } from './InventorySyncConstants';

const InventorySync = ({ data: { settingsUrl }, syncInventory }) => {
  document.title = INSIGHTS_SYNC_PAGE_TITLE;
  return (
    <IntlProvider locale={navigator.language}>
      <div className="inventory-sync">
        <h1>{__('Red Hat Inventory Sync')}</h1>
        <p>
          {__(
            "By syncing, you'll be able to know which hosts were uploaded successfuly, as your inventory, to the Red Hat cloud"
          )}
        </p>
        <p>
          {__(`1. Obtain an RHSM API token: `)}
          <a
            href="https://access.redhat.com/management/api"
            target="_blank"
            rel="noopener noreferrer"
          >
            access.redhat.com <Icon name="external-link" size="xs" />
          </a>
          <br />
          {__("2. Copy the token to 'Red Hat Cloud token' setting: ")}
          <a href={settingsUrl} target="_blank" rel="noopener noreferrer">
            {__('Red Hat Cloud token ')}
            <Icon name="external-link" size="xs" />
          </a>
          <br />
          {__(
            '3. Now you can syncronize your inventory manually using the "Sync now" button.'
          )}
        </p>
        <div>
          <Button bsStyle="primary" onClick={syncInventory}>
            {__('Sync now')}
          </Button>
        </div>
      </div>
    </IntlProvider>
  );
};

InventorySync.propTypes = {
  syncInventory: PropTypes.func.isRequired,
  data: PropTypes.shape({
    settingsUrl: PropTypes.string.isRequired,
  }).isRequired,
};

export default InventorySync;
