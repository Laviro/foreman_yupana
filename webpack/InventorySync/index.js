import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as actions from './InventorySyncActions';
import InventorySync from './InventorySync';

// map state to props
const mapStateToProps = state => ({});

// map action dispatchers to props
const mapDispatchToProps = dispatch => bindActionCreators(actions, dispatch);

// export reducers
export const reducers = {};

// export connected component
export default connect(mapStateToProps, mapDispatchToProps)(InventorySync);
