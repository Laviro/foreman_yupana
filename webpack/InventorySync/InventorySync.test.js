import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { noop } from 'patternfly-react';

import InventorySync from './InventorySync';

const fixtures = {
  render: {
    syncInventory: noop,
    data: {
      settingsUrl: 'www.example.com/settings',
    },
  },
};

describe('InventorySync', () =>
  testComponentSnapshotsWithFixtures(InventorySync, fixtures));
