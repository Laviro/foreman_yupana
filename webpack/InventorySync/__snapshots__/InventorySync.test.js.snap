// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventorySync render 1`] = `
<IntlProvider
  locale="en-US"
>
  <div
    className="inventory-sync"
  >
    <h1>
      Red Hat Inventory Sync
    </h1>
    <p>
      By syncing, you'll be able to know which hosts were uploaded successfuly, as your inventory, to the Red Hat cloud
    </p>
    <p>
      1. Obtain an RHSM API token: 
      <a
        href="https://access.redhat.com/management/api"
        rel="noopener noreferrer"
        target="_blank"
      >
        access.redhat.com 
        <Icon
          name="external-link"
          size="xs"
          type="fa"
        />
      </a>
      <br />
      2. Copy the token to 'Red Hat Cloud token' setting: 
      <a
        href="www.example.com/settings"
        rel="noopener noreferrer"
        target="_blank"
      >
        Red Hat Cloud token 
        <Icon
          name="external-link"
          size="xs"
          type="fa"
        />
      </a>
      <br />
      3. Now you can syncronize your inventory manually using the "Sync now" button.
    </p>
    <div>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="primary"
        disabled={false}
        onClick={[Function]}
      >
        Sync now
      </Button>
    </div>
  </div>
</IntlProvider>
`;
